rerun_cpp/src/rerun/archetypes/segmentation_image_ext.cpp

```diff
@@ -1,48 +1,53 @@
-#include "../error.hpp"
 #include "segmentation_image.hpp"
 
-// Uncomment for better auto-complete while editing the extension.
-// #define EDIT_EXTENSION
+#include "../collection_adapter_builtins.hpp"
+#include "../error.hpp"
 
-namespace rerun {
-    namespace archetypes {
+namespace rerun::archetypes {
 
-#ifdef EDIT_EXTENSION
-        // <CODEGEN_COPY_TO_HEADER>
+#if 0
+    // <CODEGEN_COPY_TO_HEADER>
 
-        /// New segmentation image from height/width and tensor buffer.
-        ///
-        /// Sets the dimension names to "height" and "width" if they are not specified.
-        /// Calls `Error::handle()` if the shape is not rank 2.
-        SegmentationImage(
-            std::vector<datatypes::TensorDimension> shape, datatypes::TensorBuffer buffer
-        )
-            : SegmentationImage(datatypes::TensorData(std::move(shape), std::move(buffer))) {}
+    /// New segmentation image from height/width and tensor buffer.
+    ///
+    /// Sets the dimension names to "height" and "width" if they are not specified.
+    /// Calls `Error::handle()` if the shape is not rank 2.
+    SegmentationImage(Collection<datatypes::TensorDimension> shape, datatypes::TensorBuffer buffer)
+        : SegmentationImage(datatypes::TensorData(std::move(shape), std::move(buffer))) {}
 
-        /// New segmentation image from tensor data.
-        ///
-        /// Sets the dimension names to "height" and "width" if they are not specified.
-        /// Calls `Error::handle()` if the shape is not rank 2.
-        explicit SegmentationImage(components::TensorData _data);
+    /// New segmentation image from tensor data.
+    ///
+    /// Sets the dimension names to "height" and "width" if they are not specified.
+    /// Calls `Error::handle()` if the shape is not rank 2.
+    explicit SegmentationImage(components::TensorData _data);
 
-        // </CODEGEN_COPY_TO_HEADER>
+    // </CODEGEN_COPY_TO_HEADER>
 #endif
 
-        SegmentationImage::SegmentationImage(components::TensorData _data)
-            : data(std::move(_data)) {
-            auto& shape = data.data.shape;
-            if (shape.size() != 2) {
-                Error(ErrorCode::InvalidTensorDimension, "Shape must be rank 2.").handle();
-                return;
-            }
+    SegmentationImage::SegmentationImage(components::TensorData _data) : data(std::move(_data)) {
```

Comment:
nit: we use `_prefix` for private members and `suffix_` for parameters (when needed)

```suggestion
    SegmentationImage::SegmentationImage(components::TensorData data_) : data(std::move(data_)) {
```

Reply:
missed that one, thanks! Tried to clean that thing up as I go